### PR TITLE
Update versions for 4.50

### PR DESCRIPTION
**Pull request by fredemmott**
Manually update hack array migration options documentation

Mostly changes AsyncMysqlConnectionPool(array()) to taking darray[]

### CODE REVIEW

File: guides/hack/11-built-in-types/56-darray-varray-runtime-options.md [hunk 1]
```diff
@@ -67,3 +67,13 @@ The third usecase used to be valid Hack. Unsetting the last index of a `varray<_
 The third situation, writing to a string key, is always a mistake.
 
 If this is a string literal, the actual type is most likely `darray<_, _>`. If this is an intergral string coming from an untyped function, it is worth investigating casting the value to an int.
+
+## Runtime typetests of shapes and tuples
+
+Shapes and tuples are currently implemented with `darray` and `varray`, however
+this is an implementation detail, and expected to change to `dict` and `vec`;
+this means that the following checks will currently always fail, but may
+pass in the future - f this reason, the `hhvm.hack_arr_is_shape_tuple_notices`
```
Comment (Wilfred) [inline guides/hack/11-built-in-types/56-darray-varray-runtime-options.md]: *for this reason

*type tests

File: guides/hack/11-built-in-types/56-darray-varray-runtime-options-examples/hack_arr_is_shape_tuple_notices.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?hh // strict
+
+namespace Hack\UserDocumentation\Types\VarrayDarrayRuntimeOptions\ShapeTypes;
+
+<<__EntryPoint>>
+function main(): void {
+  $_ = dict[] is shape();
+  $_ = vec[] is (int);
```
Comment (Wilfred) [inline guides/hack/11-built-in-types/56-darray-varray-runtime-options-examples/hack_arr_is_shape_tuple_notices.php]: Why the parentheses `(int)` here?

File: guides/hack/11-built-in-types/22-arrays-examples/legacy-array-fruits.php [hunk 3]
```diff
@@ -1,17 +0,0 @@
...
-function main(): void {
-  $fruits = array(); // create an empty array
-  \var_dump($fruits);
-
-  $fruits['oranges'] = 22; // create element 4 with value "black"
-  \var_dump($fruits);
-
-  $fruits['oranges'] = 18; // replace element 4’s value with "red"
-  $fruits['apples'] = 15; // create element 8 with value "white"
-  $fruits['pears'] = 12; // create element -3 with value "blue"
```
Comment (Wilfred) [inline guides/hack/11-built-in-types/22-arrays-examples/legacy-array-fruits.php]: Might be worthwhile checking that all test files are used, as part of our build.

(I'm also puzzled why this says 'element -3' here, but whatever.)

File: guides/hack/11-built-in-types/56-darray-varray-runtime-options-examples/hack_arr_is_shape_tuple_notices.php (hunk 2 above)
Comment (fredemmott) [reply guides/hack/11-built-in-types/56-darray-varray-runtime-options-examples/hack_arr_is_shape_tuple_notices.php]: it's a tuple containing an int

File: guides/hack/11-built-in-types/22-arrays-examples/legacy-array-fruits.php (hunk 3 above)
Comment (fredemmott) [reply guides/hack/11-built-in-types/22-arrays-examples/legacy-array-fruits.php]: Yeah, I deleted this test as... it makes no sense and it's unreferenced.

**Pull request merged**